05.ls/ls.rb

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'etc'
+require 'optparse'
+require 'date'
+
+def main
+  reverse_flag = list_flag = 0
+  OptionParser.new do |opt|
+    items, hidden_items = recieve_items
+    opt.on('-a') { items += hidden_items }
+    opt.on('-r') { reverse_flag = 1 }
+    opt.on('-l') { list_flag = 1 }
+    opt.parse!(ARGV)
+    sorted_items = reverse_flag == 1 ? items.sort.reverse : items.sort
```

Comment:
フラグという言葉は二つの値（今回のケースだと 0/1）のみを取ることを表しますので、`boolean`型が最適かと思います。数値だと0/1以外の値も取れてしまいますし。
現状でいう0を`false`、1を`true`に置き換えれば良さそうです。こうすると`== 1`がいらないのでスッキリかけるというメリットもあります。

```suggestion
    sorted_items = reverse_flag ? items.sort.reverse : items.sort
```

Reply:
フラグをboolean型に修正しました。